server/lib/fetch.py

```diff
@@ -0,0 +1,197 @@
+          data[var][entity] = []
+        else:
+          data[var][entity] = {}
+  result['data'] = data
+  return result
+
+
+def _compact_series(series_resp, all_facets):
+  result = {
+      'facets': {},
+  }
+  if all_facets:
+    result['facets'] = series_resp['facets']
+  data = {}
+  for var, var_obs in series_resp.get('byVariable', {}).items():
+    data[var] = {}
+    for entity, entity_obs in var_obs.get('byEntity', {}).items():
+      data[var][entity] = None
+      if 'orderedFacets' in entity_obs:
+        if all_facets:
+          data[var][entity] = [
+              _convert_v2_obs_series(x) for x in entity_obs['orderedFacets']
+          ]
+        else:
+          # There should be only one series
+          data[var][entity] = _convert_v2_obs_series(
+              entity_obs['orderedFacets'][0])
+          facet_id = data[var][entity]['facet']
+          result['facets'][facet_id] = series_resp['facets'][facet_id]
+      else:
+        if all_facets:
+          data[var][entity] = []
+        else:
+          data[var][entity] = {
+              'series': [],
+          }
+  result['data'] = data
+  return result
+
+
+def point_core(entities, variables, date, all_facets):
+  resp = dc.obs_point(entities, variables, date)
+  resp['facets'] = _get_processed_facets(resp.get('facets', {}))
+  return _compact_point(resp, all_facets)
+
+
+def point_within_core(parent_entity, child_type, variables, date, all_facets):
+  resp = dc.obs_point_within(parent_entity, child_type, variables, date)
+  resp['facets'] = _get_processed_facets(resp.get('facets', {}))
+  return _compact_point(resp, all_facets)
+
+
+def series_core(entities, variables, all_facets):
+  resp = dc.obs_series(entities, variables)
+  resp['facets'] = _get_processed_facets(resp.get('facets', {}))
+  return _compact_series(resp, all_facets)
+
+
+def series_within_core(parent_entity, child_type, variables, all_facets):
+  resp = dc.obs_series_within(parent_entity, child_type, variables)
+  resp['facets'] = _get_processed_facets(resp.get('facets', {}))
+  return _compact_series(resp, all_facets)
+
+
+def property_values(nodes, prop, out=True):
+  """Returns a compact property values data out of REST API response."""
+  resp = dc.property_values(nodes, prop, out)
+  result = {}
+  for node, node_arcs in resp.get('data', {}).items():
+    result[node] = []
+    for v in node_arcs.get('arcs', {}).get(prop, {}).get('nodes', []):
+      if 'dcid' in v:
+        result[node].append(v['dcid'])
+      else:
+        result[node].append(v['value'])
+  return result
+
+
+def raw_property_values(nodes, prop, out=True):
+  """Returns full property values data out of REST API response."""
+  resp = dc.property_values(nodes, prop, out)
+  result = {}
+  for node, node_arcs in resp.get('data', {}).items():
+    result[node] = node_arcs.get('arcs', {}).get(prop, {}).get('nodes', [])
+  return result
+
+
+def triples(nodes, out=True):
+  """Fetch triples for given nodes"""
```

Comment:
nit: add function comments for what the return object looks like for all the functions in this module